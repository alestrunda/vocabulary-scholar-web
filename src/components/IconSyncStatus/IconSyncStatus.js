import React from 'react';
import PropTypes from 'prop-types';
import CircularProgress from 'material-ui/CircularProgress';
import NotificationSync from 'material-ui/svg-icons/notification/sync';
import NotificationSyncDisabled from 'material-ui/svg-icons/notification/sync-disabled';
import IconButton from 'material-ui/IconButton';

const IconSyncStatus = ({
  className,
  iconStyle,
  isLoading,
  isSynced,
  loaderColor,
  style,
  tooltip,
  tooltipPosition,
}) => {
  return (
    <IconButton
      className={className}
      style={style}
      tooltip={tooltip}
      tooltipPosition={tooltipPosition}
    >
      {isLoading && (
        <CircularProgress
          size={30}
          color={loaderColor}
          innerStyle={{ display: 'block' }}
        />
      )}
      {!isLoading && (
        <React.Fragment>
          {isSynced && <NotificationSync color={loaderColor} />}
          {!isSynced && <NotificationSyncDisabled style={iconStyle} />}
        </React.Fragment>
      )}
    </IconButton>
  );
};

IconSyncStatus.propTypes = {
  className: PropTypes.string,
  tooltip: PropTypes.string,
  tooltipPosition: PropTypes.string,
  isLoading: PropTypes.bool,
  isSynced: PropTypes.bool,
  iconStyle: PropTypes.object,
  loaderColor: PropTypes.string,
  style: PropTypes.object,
};

IconSyncStatus.defaultProps = {
  tooltipPosition: 'bottom-center',
};

export default IconSyncStatus;
