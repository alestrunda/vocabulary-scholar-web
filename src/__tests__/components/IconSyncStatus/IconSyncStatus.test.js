import React from 'react';
import { shallow } from 'enzyme';

import IconSyncStatus from '../../../components/IconSyncStatus';

it('renders correctly', () => {
  const root = shallow(<IconSyncStatus />);
  expect(root).toMatchSnapshot();
});
