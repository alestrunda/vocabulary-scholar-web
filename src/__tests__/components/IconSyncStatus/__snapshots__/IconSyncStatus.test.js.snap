// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IconSyncStatus
    tooltipPosition="bottom-center"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        undefined,
        <React.Fragment>
          <NotificationSyncDisabled />
        </React.Fragment>,
      ],
      "className": undefined,
      "disableTouchRipple": false,
      "disabled": false,
      "iconStyle": Object {},
      "style": undefined,
      "tooltip": undefined,
      "tooltipPosition": "bottom-center",
      "touch": false,
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            undefined,
            <NotificationSyncDisabled />,
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "style": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          undefined,
          <React.Fragment>
            <NotificationSyncDisabled />
          </React.Fragment>,
        ],
        "className": undefined,
        "disableTouchRipple": false,
        "disabled": false,
        "iconStyle": Object {},
        "style": undefined,
        "tooltip": undefined,
        "tooltipPosition": "bottom-center",
        "touch": false,
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              undefined,
              <NotificationSyncDisabled />,
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "style": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
